clients/client-py/taskcluster/helper.py

```diff
@@ -128,3 +144,44 @@ def load_secrets(
             raise Exception("Missing value {} in secrets.".format(required_secret))
 
     return secrets
+
+
+def upload_artifact(queue_service, artifact_path, content, content_type, ttl):
+    """
+    Create an artifact on the current Taskcluster Task in 2 steps:
+    1. create the Artifact through the API
+    2. upload the file on the storage provider
+    """
+    task_id = os.environ.get("TASK_ID")
+    run_id = os.environ.get("RUN_ID")
+    proxy = os.environ.get("TASKCLUSTER_PROXY_URL")
+    assert task_id and run_id and proxy, "Can only run in Taskcluster tasks with proxy"
+    assert isinstance(content, str)
+    assert isinstance(ttl, datetime.timedelta)
+
+    # Create S3 artifact on Taskcluster
+    resp = queue_service.createArtifact(
+        task_id,
+        run_id,
+        artifact_path,
+        {
+            "storageType": "s3",
+            "expires": stringDate(datetime.datetime.utcnow() + ttl),
+            "contentType": content_type,
+        },
+    )
+    assert resp["storageType"] == "s3", "Not an s3 storage"
+    assert "putUrl" in resp, "Missing putUrl"
+    assert "contentType" in resp, "Missing contentType"
+
+    # Push the artifact on storage service
+    headers = {"Content-Type": resp["contentType"]}
+    push = requests.put(url=resp["putUrl"], headers=headers, data=content)
+    push.raise_for_status()
+
+    # Build the absolute url
+    return "/api/queue/v1/task/{task_id}/runs/{run_id}/artifacts/{path}".format(
```

Comment:
My only question is if `path` can contain extra/preceding slashes. I suppose if it does, the artifact url would likely also contain those extra slashes.

Reply:
Is it possible to create an artifact on taskcluster with a leading `/` ?
Maybe I can add an assertion to avoid that ?

Reply:
It's possible, but inadvisable.  I suspect it would "just work" with this implementation, so probably no need for an assertion.  I'll get this merged :)